### ISSUE TITLE
[Introspection] Failed to parse the schema it generated.

### ISSUE THREAD

**Issue by joshuaavalon** [prisma/prisma2#1632]
Report ID: `2080`

I was trying out Primsa2. I created PostgresSQL database via Docker.

The database is initialized with the following SQL.

```sql
CREATE EXTENSION IF NOT EXISTS "uuid-ossp";

CREATE TABLE "public"."User" (
  id UUID PRIMARY KEY DEFAULT uuid_generate_v1(),
  name VARCHAR(255) UNIQUE NOT NULL,
  email VARCHAR(255) UNIQUE NOT NULL,
  password VARCHAR(255) NOT NULL
);

CREATE TABLE "public"."TagCategory" (
  id UUID PRIMARY KEY DEFAULT uuid_generate_v1(),
  name VARCHAR(255) UNIQUE NOT NULL
);

CREATE TABLE "public"."Tag" (
  id UUID PRIMARY KEY DEFAULT uuid_generate_v1(),
  name VARCHAR(255) UNIQUE NOT NULL,
  description TEXT NOT NULL,
  category_id UUID NOT NULL,
  FOREIGN KEY (category_id) REFERENCES "public"."TagCategory"(id)
);

CREATE TABLE "public"."TagRelationship" (
  id UUID NOT NULL,
  FOREIGN KEY (id) REFERENCES "public"."Tag"(id),
  relate_id UUID NOT NULL,
  FOREIGN KEY (relate_id) REFERENCES "public"."Tag"(id),
  PRIMARY KEY (id, relate_id),
  parent BOOLEAN NOT NULL
);

CREATE TABLE "public"."GalleryCategory" (
  id UUID PRIMARY KEY DEFAULT uuid_generate_v1(),
  name VARCHAR(255) UNIQUE NOT NULL
);

CREATE TABLE "public"."Gallery" (
  id UUID PRIMARY KEY DEFAULT uuid_generate_v1(),
  created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
  title VARCHAR(255) NOT NULL,
  description TEXT NOT NULL,
  author_id UUID NOT NULL,
  FOREIGN KEY (author_id) REFERENCES "public"."User"(id),
  category_id UUID NOT NULL,
  FOREIGN KEY (category_id) REFERENCES "public"."GalleryCategory"(id)
);

CREATE TABLE "public"."GalleryTag" (
  tag_id UUID NOT NULL,
  FOREIGN KEY (tag_id) REFERENCES "public"."Tag"(id),
  gallery_id UUID NOT NULL,
  FOREIGN KEY (gallery_id) REFERENCES "public"."Gallery"(id),
  PRIMARY KEY (tag_id, gallery_id)
);

CREATE TABLE "public"."Image" (
  id UUID PRIMARY KEY DEFAULT uuid_generate_v1(),
  position INT NOT NULL,
  gallery_id UUID NOT NULL,
  FOREIGN KEY (gallery_id) REFERENCES "public"."Gallery"(id),
  UNIQUE (position, gallery_id)
);

```

I tried to run `npx prisma2 introspect` with

```prisma
datasource db {
  provider = "postgresql"
  url      = "postgresql://admin:admin@192.168.1.13:5432/admin"
}

generator client {
  provider = "prisma-client-js"
}
```

It prints the following log.
```
Introspecting based on datasource defined in schema.prisma …

Introspection failed: Introspected schema can't be parsed.
Introspected Schema:

datasource db {
  provider = "postgresql"
  url      = "postgresql://admin:admin@192.168.1.13:5432/admin"
}
model Gallery {
  created_at  DateTime
  description String
  id          String          @id
  title       String
  author_id   User
  category_id GalleryCategory
  galleryTags GalleryTag[]    @relation(references: [gallery_id])
  images      Image[]         @relation(references: [gallery_id])
}

model GalleryCategory {
  id        String    @id
  name      String    @unique
  galleries Gallery[]
}

model GalleryTag {
  gallery_id Gallery
  tag_id     Tag

  @@id([tag_id, gallery_id])
}

model Image {
  id         String  @id
  position   Int
  gallery_id Gallery

  @@unique([position, gallery_id], name: "Image_position_gallery_id_key")
}

model Tag {
  description                                     String
  id                                              String            @id
  name                                            String            @unique
  category_id                                     TagCategory
  galleryTags                                     GalleryTag[]
  tagRelationships_TagToTagRelationship_id        TagRelationship[] @relation("TagToTagRelationship_id", references: [id])
  tagRelationships_TagToTagRelationship_relate_id TagRelationship[] @relation("TagToTagRelationship_relate_id", references: [relate_id])
}

model TagCategory {
  id   String @id
  name String @unique
  tags Tag[]
}

model TagRelationship {
  parent    Boolean
  id        Tag     @relation("TagToTagRelationship_id")
  relate_id Tag     @relation("TagToTagRelationship_relate_id")

  @@id([id, relate_id])
}

model User {
  email     String    @unique
  id        String    @id
  name      String    @unique
  password  String
  galleries Gallery[]
}

Schema parsing
We successfully received the error report
To help us even more, please create an issue at https://github.com/prisma/prisma2/issues/new
mentioning the report id 2080.

Thanks a lot for your help! �🙏
```

I did not craft the models but Prisma seems to fail to parsed the schema it generates.

**Comment by janpio** [comment]
Thanks @joshuaavalon. 

I see you are running `2.0.0-preview021` via the error report.

This should already be fixed in the lastest `alpha`, can you maybe try upgrading or running via `npx` (`npx prisma2@alpha introspect` and then `npx prisma2@alpha generate`) to verify? 

This change will also be included in our next preview release `2.0.0-preview022` of course (which will drop later today).

**Comment by joshuaavalon** [comment]
Thanks @janpio . It seems to work fine in Preview022. I will try it out more.
